### ISSUE TITLE
Hang-up with footnote or page float on pseudo elements

### ISSUE THREAD

**Issue by MurakamiShinyu** [vivliostyle/vivliostyle.js#703]
**Describe the bug**

Vivliostyle hangs up when `float: footnote` is specified on before/after pseudo elements.

**To Reproduce**

Test: https://jsbin.com/nolaxej

```html
    <style>
      a::after {
        float: footnote;
        content: "(" attr(href) ")";
      }
    </style>

    <h1>Test: Footnote on pseudo element</h1>
    <p>See <a href="https://example.com">Example</a>.</p>
```

Test with Vivliostyle Viewer: https://vivliostyle.org/viewer/#src=https://jsbin.com/nolaxej&debug=true (with debug flag to see what's happening)

**Comment by MurakamiShinyu** [comment]
Same problem with page floats.

Test: https://jsbin.com/texokix
with Vivliostyle Viewer: https://vivliostyle.org/viewer/#src=https://jsbin.com/texokix&debug=true

**Comment by MurakamiShinyu** [comment]
Fixed in PR #704.

Test with the canary version: 
- https://vivliostyle.now.sh/#src=https://jsbin.com/nolaxej
- https://vivliostyle.now.sh/#src=https://jsbin.com/texokix

Also test with the printcss example by @azettl, thanks for making me aware of this bug:
- https://vivliostyle.now.sh/#src=https://github.com/azettl/printcss.examples/blob/main/HTML%20Examples/Books%20and%20Magazines/howcolors.work/index.html
